test/unit/RedisClusterOptionsTest.php

```diff
@@ -38,6 +38,7 @@ public function testCanHandleOptionsWithNodename(): void
             'persistent'    => false,
             'redis_version' => '1.0',
             'password'      => 'secret',
+            'ssl_context'   => ['verify_peer' => false],
```

Comment:
Lets allow instantiation from array shape and object (which needs to be implemented).
Allowing an array with whatever key might introduce problems if keys are passed as `ssl_context` which are not allowed. If we can prevent that from static analysis level, I'd say lets do so.